File: 1.two-sum.cpp
/*
 * @lc app=leetcode id=1 lang=cpp
 *
 * [1] Two Sum
 */
#include <bits/stdc++.h>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> seen;

        for (int i = 0; i < nums.size(); ++i)
        {
            int complement = target - nums[i];

            if (seen.count(nums[i]))
            {
                return {seen[nums[i]], i};
            }
            else
            {
                seen[complement] = i;
            }
        }

        return {-1, -1};
    }
};
// @lc code=end

File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#
from typing import *
# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        
        for i in range(len(nums)):
            remainder = target - nums[i]
            if remainder in map:
                return [i, map[remainder]]
            
            map[nums[i]] = i
        
        return [-1, -1]
            
# @lc code=end

